instaclustr/resource_cluster.go

```diff
@@ -401,18 +668,36 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 
 	clusterProvider.Tags = d.Get("tags").(map[string]interface{})
 
-	createData := CreateRequest{
+	var createData = CreateRequest{
 		ClusterName:           d.Get("cluster_name").(string),
 		Bundles:               bundles,
-		Provider:              clusterProvider,
+		Provider:              &clusterProvider,
 		SlaTier:               d.Get("sla_tier").(string),
 		NodeSize:              d.Get("node_size").(string),
-		DataCentre:            d.Get("data_centre").(string),
-		ClusterNetwork:        d.Get("cluster_network").(string),
 		PrivateNetworkCluster: fmt.Sprintf("%v", d.Get("private_network_cluster")),
 		PCICompliantCluster:   fmt.Sprintf("%v", d.Get("pci_compliant_cluster")),
 	}
 
+	dataCentre := d.Get("data_centre").(string)
+	dataCentres, err := getDataCentres(d)
+	if err != nil {
+		return formatCreateErrMsg(err)
+	}
+
+	// we will throw an error if neither data_centre nor data_centres found
+	if dataCentre == "" && len(dataCentres) == 0 {
+		return fmt.Errorf("[Error] Error creating cluster: either data_centre or data_centres should be provided")
+	}
+
+	// for a single DC cluster
+	if dataCentre != "" && len(dataCentres) == 0 {
```

Comment:
[desired] This check is used at least twice, maybe it could be made into a function to increase readability and reuse. Not required

Reply:
Also can dataCentre be `nil` here?

Reply:
resolved